parser.ts

```diff
@@ -219,8 +222,16 @@ export class Parser {
   private advance(): Token {
     this.previous = this.current;
     this.current = this.next;
-    this.line += this.nextWs + this.current.text;
+
+    if (this.line.lineNo.length && !this.line.rest.length) {
+      this.line.trailingWhitespace = this.nextWs;
+    } else {
+      this.line.rest += this.nextWs;
+    }
+    this.line.rest += this.current.text;
+
     const [ws, next] = this.nextToken();
+    this.leadingWs = this.nextWs;
```

Comment:
This might be suspect, but I'd expect bugs here to manifest in the wrong amount of whitespace. As an aside, coverage here is... not great.